corehq/apps/locations/permissions.py

```diff
@@ -98,3 +102,22 @@ def _inner(request, domain, *args, **kwargs):
             return view_fn(request, domain, *args, **kwargs)
         raise Http404()
     return locations_access_required(_inner)
+
+
+def can_edit_form_location(domain, user, instance_id):
+    if not toggles.RESTRICT_FORM_EDIT_BY_LOCATION.enabled(domain):
+        return True
+
+    location = get_xform_location(XFormInstance.get(instance_id))
```

Comment:
also, this pattern is nice in the sense that the decorator is easy and intuitive to add to the views. however the downside is now every view that succeeds (i guess only if the toggle is on) will have to get the form twice unecessarily. it might be nice to actually build this function into the `_get_form_or_404` that all (or at least) most of the decorated views use, that way we still only have to get the doc once.